src/main/java/com/google/cloud/tools/appengine/cloudsdk/PathResolver.java

```diff
@@ -85,8 +94,33 @@ private static void getLocationsFromPath(List<String> possiblePaths) {
         if (path.endsWith("google-cloud-sdk" + File.separator + "bin")) {
           possiblePaths.add(path.substring(0, path.length() - 4));
         }
+
+        Path possibleLink = Paths.get(path, "gcloud");
+        if (Files.isSymbolicLink(possibleLink)) {
+          getLocationsFromLink(possiblePaths, possibleLink);
+        }
+      }
+    }
+  }
+
+  @VisibleForTesting
+  // resolve symlinks to a path that could be the bin directory of the cloud sdk
+  static void getLocationsFromLink(List<String> possiblePaths, Path link) {
+    try {
+      Path resolvedLink = link.toRealPath();
+      Path possibleBinDir = resolvedLink.getParent();
```

Comment:
> But then, the last portion of resolvedLink is always gcloud, so I guess this is OK.

Actually, it could matter if the path was like `/home/google-cloud-sdk/bin/./gcloud`. `possibleBinDir.getFileName()` will get you `.` instead of `bin`.